api/src/main/java/io/minio/MinioClient.java

```diff
@@ -928,7 +921,15 @@ private Request createRequest(Method method, String bucketName, String objectNam
       // Handle putobject specially to use chunked upload.
       if (method == Method.PUT && objectName != null && body != null && body instanceof InputStream && length > 0) {
         sha256Hash = "STREAMING-AWS4-HMAC-SHA256-PAYLOAD";
-        requestBuilder.header("Content-Encoding", "aws-chunked");
+
+        if (headerMap != null) {
+          requestBuilder.header("Content-Encoding", Stream.concat(Stream.of("aws-chunked"),
+              headerMap.get("Content-Encoding").stream()).distinct().filter(encoding -> !encoding.isEmpty())
+              .collect(Collectors.joining(",")));
+        } else {
+          requestBuilder.header("Content-Encoding", "aws-chunked");
```

Comment:
You could do 
```java
String contentEncoding = "aws-chunked";
if (headerMap != null) {
  contentEncoding = Stream.concat(Stream.of("aws-chunked"),
              headerMap.get("Content-Encoding").stream()).distinct().filter(encoding -> !encoding.isEmpty())
              .collect(Collectors.joining(","));
}
requestBuilder.header("Content-Encoding", contentEncoding)
```

Reply:
Yeah, looks good. I suppose I will wait for all the comments to incorporate the changes.

Reply:
Rest of things look good.

Reply:
Done